Fix the automated release process

# Problems

1. The release-if-necessary Github Workflow fails. [Example](https://github.com/RequestNetwork/web-components/actions/runs/9276585203). It should automatically deploy packages to NPM when their version number increases.
2. Upgrading inter-package dependencies is prone to human error

# Definition of Done

- [x] Fix the `release-if-necessary` Github Workflow. 
- [x] ~~Automate upgrading inter-package dependencies so they're always using the latest versions of each package.~~ Unnecessary after #41

Unfortunately, the release process is still broken and we're unsure why.
https://github.com/RequestNetwork/web-components/actions/runs/9944543070

<details><summary>Publish logs</summary>
<p>

```
Run npm config set registry https://registry.npmjs.org/
  npm config set registry https://registry.npmjs.org
  npm publish --workspace=@requestnetwork/create-invoice-form
  shell: /usr/bin/bash -e {0}
  env:
    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
    NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX

> @requestnetwork/create-invoice-form@0.6.0 prepublishOnly
> npm run package


> @requestnetwork/create-invoice-form@0.6.0 package
> svelte-kit sync && svelte-package && npm run build:wc && publint

Your compilerOptions.paths in tsconfig.json should include the following:
{
  "$lib":["./src/lib"],
  "$lib/*":["./src/lib/*"]
}
src/lib -> dist
@sveltejs/package found the following issues while packaging your library:
You are using Svelte components or Svelte-specific imports in your code, but you have not declared a dependency on `svelte` in your `package.json`. Add it to your `dependencies` or `peerDependencies`.


> @requestnetwork/create-invoice-form@0.6.0 build:wc
> vite build -c vite.wc.config.ts

vite v4.5.3 building for production...
transforming...
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/packages/create-invoice-form/src/lib/invoice/form.svelte:470:[2](https://github.com/RequestNetwork/web-components/actions/runs/9944543070/job/27470818778#step:8:2) Unused CSS selector "button"
468:   }
469: 
470:   button {
       ^
471:     background: transparent;
472:     border: none;
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/packages/create-invoice-form/src/lib/invoice/form.svelte:531:2 Unused CSS selector ".invoice-form-dates input"
529:   }
530: 
531:   .invoice-form-dates input {
       ^
532:     width: 100%;
533:   }
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/shared/components/dropdown.svelte:42:2 A11y: visible, non-interactive elements with an on:click event must be accompanied by a keyboard event handler. Consider whether an interactive element such as <button type="button"> or <a> might be more appropriate. See https://svelte.dev/docs/accessibility-warnings#a11y-click-events-have-key-events for more details.
40: 
41: {#if isOpen}
42:   <div class="dropdown-overlay" on:click={closeDropdown}></div>
      ^
43: {/if}
44: 
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/shared/components/dropdown.svelte:42:2 A11y: <div> with click handler must have an ARIA role
40: 
41: {#if isOpen}
42:   <div class="dropdown-overlay" on:click={closeDropdown}></div>
      ^
43: {/if}
44: 
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/shared/components/dropdown.svelte:170:2 Unused CSS selector ".dropdown-menu > * + *"
168:   }
169: 
170:   .dropdown-menu > * + * {
       ^
171:     border-top-width: 1px;
172:     border-bottom-width: 0px;
6:19:35 PM [vite-plugin-svelte] /home/runner/work/web-components/web-components/shared/components/accordion.svelte:63:2 Unused CSS selector ".accordion-close-icon"
61:   }
62: 
63:   .accordion-close-icon {
      ^
64:     transform: rotate(180deg);
65:   }
[plugin:vite:resolve] Module "crypto" has been externalized for browser compatibility, imported by "/home/runner/work/web-components/web-components/node_modules/@requestnetwork/utils/dist/crypto/crypto-wrapper.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "crypto" has been externalized for browser compatibility, imported by "/home/runner/work/web-components/web-components/node_modules/@toruslabs/eccrypto/dist/eccrypto.esm.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
✓ 2556 modules transformed.
rendering chunks...
computing gzip size...
dist/web-component.js  4,940.76 kB │ gzip: 1,125.39 kB │ map: 11,889.84 kB
dist/web-component.umd.cjs  3,654.75 kB │ gzip: 988.80 kB │ map: 11,522.07 kB
✓ built in 16.66s
@requestnetwork/create-invoice-form lint results:
All good!
npm notice
npm notice 📦  @requestnetwork/create-invoice-form@0.6.0
npm notice Tarball Contents
npm notice 6.1kB README.md
npm notice 7.6kB dist/create-invoice-form.svelte
npm notice 795B dist/create-invoice-form.svelte.d.ts
npm notice 394.1kB dist/fonts/Montserrat-VariableFont_wght.ttf
npm notice 77B dist/index.d.ts
npm notice 77B dist/index.js
npm notice 11.2kB dist/invoice/form-view.svelte
npm notice 890B dist/invoice/form-view.svelte.d.ts
npm notice 19.6kB dist/invoice/form.svelte
npm notice 903B dist/invoice/form.svelte.d.ts
npm notice 117B dist/invoice/index.d.ts
npm notice 117B dist/invoice/index.js
npm notice 107B dist/utils/index.d.ts
npm notice 107B dist/utils/index.js
npm notice 502B dist/utils/prepareRequest.d.ts
npm notice 3.9kB dist/utils/prepareRequest.js
npm notice 1.3kB dist/utils/resetForm.d.ts
npm notice 1.4kB dist/utils/resetForm.js
npm notice 4.9MB dist/web-component.js
npm notice 11.9MB dist/web-component.js.map
npm notice 3.7MB dist/web-component.umd.cjs
npm notice 11.5MB dist/web-component.umd.cjs.map
npm notice 1.5kB package.json
npm notice Tarball Details
npm notice name: @requestnetwork/create-invoice-form
npm notice version: 0.6.0
npm notice filename: requestnetwork-create-invoice-form-0.6.0.tgz
npm notice package size: 7.3 MB
npm notice unpacked size: 32.5 MB
npm notice shasum: a2e89364c40fff9e05dd7070ed435904ebcd78f6
npm notice integrity: sha512-z5sMw7qi2iutH[...]TI1uk+gLa/kPQ==
npm notice total files: 23
npm notice
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access
npm error code E404
npm error 404 Not Found - PUT https://registry.npmjs.org/@requestnetwork%2fcreate-invoice-form - Not found
npm error 404
npm error 404  '@requestnetwork/create-invoice-form@0.6.0' is not in this registry.
npm error 404
npm error 404 Note that you can also install from a
npm error 404 tarball, folder, http url, or git url.

npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2024-07-15T18_19_28_58[3](https://github.com/RequestNetwork/web-components/actions/runs/9944543070/job/27470818778#step:8:3)Z-debug-0.log
Error: Process completed with exit code 1.
```

</p>
</details>

--- a/.github/workflows/npm-publish.yaml
+++ b/.github/workflows/npm-publish.yaml
@@ -1,3 +1,5 @@
+# @format
+
 name: release-if-necessary
 
 on:
@@ -12,9 +14,9 @@ jobs:
     strategy:
       matrix:
         package:
-          - "@requestnetwork/add-stakeholder"
-          - "@requestnetwork/create-invoice-form"
-          - "@requestnetwork/invoice-dashboard"
+          - '@requestnetwork/add-stakeholder'
+          - '@requestnetwork/create-invoice-form'
+          - '@requestnetwork/invoice-dashboard'
     steps:
       - name: Checkout repository 🛎️
         uses: actions/checkout@v4
@@ -22,8 +24,8 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: "18"
-          registry-url: "https://registry.npmjs.org"
+          node-version: '18'
+          registry-url: 'https://registry.npmjs.org'
 
       - name: Install dependencies 📥
         run: npm ci
@@ -39,14 +41,9 @@ jobs:
           echo "is-release-needed=$IS_RELEASE_NEEDED"
           echo "is-release-needed=$IS_RELEASE_NEEDED" >> $GITHUB_OUTPUT
 
-      - name: Setup .npmrc file to publish to npm 📝
-        if: steps.is-release-needed.outputs.is-release-needed == 'true'
-        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.REQUEST_BOT_NPM_TOKEN }}" > ~/.npmrc
-
       - name: Publish package on NPM 📦
         if: steps.is-release-needed.outputs.is-release-needed == 'true'
         run: |
-          npm config set registry https://registry.npmjs.org
-          npm whoami || { echo "Failed to identify npm user."; exit 1; }
-          npm config get registry || { echo "Failed to get npm registry."; exit 1; }
           npm publish --workspace=${{ matrix.package }} --dry-run
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.REQUEST_BOT_NPM_TOKEN }}

